bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/FlatRowAdapter.java

```diff
@@ -82,23 +122,38 @@ private Result convert(FlatRow flatRow, Collection<org.apache.hadoop.hbase.Cell>
       // HBase will treat them as duplicates.
       long hbaseTimestamp = cell.getTimestamp() / TIME_CONVERSION_UNIT;
       byte[] family = getFamily(previousCell, previousFamily, cell);
-      RowCell keyValue = new RowCell(
-          rowKey,
-          family,
-          ByteStringer.extract(cell.getQualifier()),
-          hbaseTimestamp,
-          ByteStringer.extract(cell.getValue()));
-
-      hbaseCells.add(keyValue);
+      RowCell rowCell =
+          new RowCell(
+              rowKey,
+              family,
+              ByteStringer.extract(cell.getQualifier()),
+              hbaseTimestamp,
+              ByteStringer.extract(cell.getValue()));
+      if (!sort ) {
+        hbaseCells.add(rowCell);
+      } else {
+        if (previousFamily != null && family != previousFamily) {
+          addAll(hbaseCells, familyCells);
+        }
+        familyCells.add(rowCell);
+      }
       previousCell = cell;
       previousFamily = family;
     }
+    addAll(hbaseCells, familyCells);
+    return hbaseCells;
+  }
 
-    return Result.create(hbaseCells.toArray(new org.apache.hadoop.hbase.Cell[hbaseCells.size()]));
+  void addAll(List<Cell> hbaseCells, List<RowCell> familyCells) {
```

Comment:
nit: Rearrange parameter ordering?  void addAll(List<RowCell> familyCells, List<Cell> hbaseCells, )
Usually, we order the input as "from" and then "to". "Add cells from 'familyCells' to 'hbaseCells'".